feat: ledger integration

stanleyyconsensys opened the pull request:
## **Description**
_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

## **Manual testing steps**

_1. Step1:_
_2. Step2:_
_3. ..._

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

joaoloureirop committed:
(util/address): remove duplicate import

joaoloureirop committed:
Merge branch 'main' into feature/4754-permission-system-bugbash-build

Cal-L committed:
Merge from main

sethkfman committed:
updated android SDK versions

joaoloureirop committed:
update to react-native 0.66.5

more info https://github.com/facebook/react-native/issues/35210

andrepimenta committed:
SDK compatibility

andrepimenta committed:
Update coinbase

Cal-L committed:
Clean up SDK support a bit

Cal-L committed:
Add padding to TagURL

joaoloureirop committed:
fix linter

gantunesr committed:
Solve conflicts

joaoloureirop committed:
Merge branch 'main' into feature/4754-permission-system

joaoloureirop committed:
fix typo

Cal-L committed:
Fix revoke button colors (#5255)

* Fix revoke button colors

* Remove unused import

Cal-L committed:
Hide tab bar on Android (#5261)

joaoloureirop committed:
Merge branch 'main' into feature/4754-permission-system

joaoloureirop committed:
fix lint errors

jpcloureiro committed:
update Permission System translations (#5401)

translations fetched from commit bda3e4a

jpcloureiro committed:
[PS] Allow Permission Request Rejection (#5405)

* get permission request id on AccountConnect

* refactor account connect action handlers

* move USER_INTENT enum to constants folder

* clenup

- remove eslint ignores
- remove old TODOs
- add useEffect comment to address why we need the user intent state
  machine

joaoloureirop committed:
(RootRPCMethodsUI): relocate requestData destruct

on Permission System we use a Permission Request ID that
is only relevant for the Permission Controller and not the other
Approvals.

Destructuring requestData to get said id should only happen on
REQUEST_PERMISSION Approval Type.

joaoloureirop committed:
Merge branch 'main' into feature/4754-permission-system

jpcloureiro committed:
move backup wallet reminder popup (#5402)

so it does not block the navigation bottom bar

gantunesr committed:
Solve conflicts

gantunesr committed:
Merge branch 'main' of https://github.com/MetaMask/metamask-mobile into feat/ledger-integration

Cal-L committed:
Fix bug report 12 & 3

Cal-L committed:
Merge from main

Cal-L committed:
Use MetaMetrics and update snapshots

joaoloureirop committed:
(Loader): add color prop

joaoloureirop committed:
remove inline styles

joaoloureirop committed:
(TabBar): remove null label condition.

This condition is not needed anymore since we removed
the Activity Route from the bottom navigation stack

Cal-L committed:
Fix snapshots

Cal-L committed:
Merge branch 'feature/4754-permission-system' of https://github.com/MetaMask/metamask-mobile into feature/4754-permission-system

Cal-L committed:
Merge with main

Cal-L committed:
Improve logic to autoscroll account selector

Cal-L committed:
Fix analytics

Cal-L committed:
Enable ability to delete imported account while in connect mode

Cal-L committed:
Auto scroll multiselect when possible

Cal-L committed:
Fix lint

Cal-L committed:
Merge with main

Cal-L committed:
Replace @metamask/controllers imports

joaoloureirop committed:
(Navigation): adopt useNavigationDetails pattern

Cal-L committed:
Improve wallet tab contrast

Cal-L committed:
Merge branch 'feature/4754-permission-system' of https://github.com/MetaMask/metamask-mobile into feature/4754-permission-system

Cal-L committed:
Remove unused network prop from browser tab

joaoloureirop committed:
force permitted accounts to be lower case

Cal-L committed:
Change favicon size to 50. Opensea doesn't support 64

Cal-L committed:
Update revoke translations

Cal-L committed:
Merge with main

Cal-L committed:
Fix lint issue

Cal-L committed:
Fix missing account name in toast

Cal-L committed:
Only allow removing imported wallets via seed

Cal-L committed:
Merge in main

Cal-L committed:
Improve avatar network placeholder padding

Cal-L committed:
Improve network icon display when switching networks

cortisiko committed:
E2e/permission system (#5539)

* fix first set of broken tests

* update tokens text

* fix deep linking assertion

Cal-L committed:
Merge with main

cortisiko committed:
fix snapshot

Cal-L committed:
Fix storybook

Cal-L committed:
Merge branch 'feature/4754-permission-system' of https://github.com/MetaMask/metamask-mobile into feature/4754-permission-system

Cal-L committed:
Fix XS avatar alignment

Cal-L committed:
Lower font size for XS avatar fallback

Cal-L committed:
Use test network images

Cal-L committed:
Keep import flow persistent within account selection

Cal-L committed:
Reduce picker network to 1 line

Cal-L committed:
Re-add skip button for onboarding wizard

gantunesr committed:
Merge main

Cal-L committed:
Clean up network icon on network info screen when switching networks

Cal-L committed:
Show toast when switching between networks with same chainId but different type

cortisiko committed:
fix onboarding tutorial appium test

gantunesr committed:
Update controllers versions

gantunesr committed:
Merge branch 'main' of https://github.com/MetaMask/metamask-mobile into feat/ledger-integration

cortisiko committed:
fix appium tests

cortisiko committed:
forgot a commit to the helpers file.

gantunesr committed:
Update dependencies

joaoloureirop committed:
Merge branch 'main' into feature/4754-permission-system

gantunesr committed:
Merge branch 'main' of https://github.com/MetaMask/metamask-mobile into feat/ledger-integration

gantunesr committed:
Add patch to @ledgerhq/react-native-hw-transport-ble

Cal-L committed:
Merge from mina

cortisiko committed:
fix broken wallet.steps

gantunesr committed:
Solve conflicts

gantunesr committed:
Update controllers preview package

gantunesr committed:
Fix breaking changes

Cal-L committed:
Handle adding and removing accounts during connect flows

Cal-L committed:
Merge branch 'feature/4754-permission-system' of https://github.com/MetaMask/metamask-mobile into feature/4754-permission-system

Cal-L committed:
Enable passing interacting address to watch asset

gantunesr committed:
Merge permissions system

gantunesr committed:
Update Routes

gantunesr committed:
Update Routes

gantunesr committed:
Update Routes

cortisiko committed:
fix detox wallet e2e

jpcloureiro committed:
Permission System UI bugfixes (#5603)

* fix network casing name in educational network info modal

* fix no network avatar for frequent networks

gantunesr committed:
Merge branch 'main' of https://github.com/MetaMask/metamask-mobile into feat/ledger-integration

SamuelSalas committed:
E2e appium/fix browser test permission system (#5611)

* updated browser tests to run on thje permission system

* Returning values to original state

* Solve account approval modal issues and option menu test issues

* Update network text value

* updating failing tests

* Merge latest changes an returning config values to default

* Update

* Add AddContact.js file

* update AddContact page object + fix linting issue

---------

Co-authored-by: Curtis <Curtis.David7@gmail.com>

cortisiko committed:
Fix E2E merge conflicts on Permission System Branch (#5635)

* [IMPROVEMENT] Refactor ProtectYourWallet section in the security section (#5600)

* [FIX] Analytics event interface (#5613)

* Fix analytics events

* Bump version and build numbers

* Revert "Bump version and build numbers"

This reverts commit 247789b8428ad7f3bd3cca858c7818ffc5a7a68a.

* Handle ENS Address Error (#5605)

* resolving ens address

* lint

* cleaner name

* Bump versions to make prod build for testing

* Revert "Bump versions to make prod build for testing"

This reverts commit 43d0dad69582f5b3d467e4a37e9dd118cb36016a.

Co-authored-by: Plasma Corral <32695229+plasmacorral@users.noreply.github.com>

* [FIX] Make Prettier work in wdio directory (#5612)

* make Prettier work in wdio directory

- remove wdio from the Prettier exclude file
- add Gherkin Prettier pluggin for .feature
- update `yarn format` script to take .feature files into account
- first pass of js and feature files formating in wdio dir using `yarn format`

* format after branch update from main

* extract eslint config to deal with it in another PR

- create an eslint ignore file and use it as default instead of prettier ignore with eslint (on both scripts, lint and fix)
- add the wdio dir back in .eslintignore to deal with it in another eslint only PR
- remove useless ignore lines as already ignored by the android line

* [FIX] SRP Quiz content and translations (#5626)

* [IMPROVEMENT] Refactor Personal Signature (#5551)

* personal signature refactor

* Fix currency display (#5623)

* [FEAT] Add friction to SRP reveal (#5579)

* Add base modals

* Add navigation for dev purposes

* Update Main Nav

* Update navigation

* Add new Route

* Update quiz files

* Update modals

* Add QuizStage enum

* Add interfaces

* Update styles

* Update quiz components

* Update IQuizInformationProps interface

* Rename files

* Add quiz flow

* Revert AccountView changes

* Add Quiz to security settings

* Update title interface and style

* Update QuizContent

* Update styles

* Update dep arrays

* Add quiz structure

* Move QuizStage to type file

* Update QuizContent component

* Add method to dismiss modal

* Revert Podfile.lock

* Fix typo

* Remove unnecessary comments

* Fix structure error

* Add analytic events

* Fix small content issues

* Add image to quiz intro

* Add image to quiz intro

* Add link to zendesk article

* Remove quiz structure cobject

* Bump version and build number

* Ignore advisories

* Revert "Bump version and build number"

This reverts commit b9833bb2db5430d99985c3db269cbc30b695746f.

* Revert "Ignore advisories"

This reverts commit 17d68ebf7ba9b684425226eae3225e2c8631e25e.

* Remove message

* Update dependencies array

* Use Routes for RevealPrivateCredentialView

* Remove unused interfaces

* Move SRP guide method to QuizModal

* Update dependency array

* Refactor switch-case statement

* Fix nav bug

* Dismiss Quiz if the app goes to the background

* E2e appium folder structure change (#5632)

* Re-arranged wdio file structure

Re-arranged wdio file structure & refactor imports

* Added missing Address Input ID

* Commented out failing test

* Removed ignored test

* rename folder to "Accounts"

* move lockResetPassword out of onboarding folder

---------

Co-authored-by: Curtis <Curtis.David7@gmail.com>

* fix import path

* fix e2e after merging main.

* fix linting

* Remove unused files. Fix personal sign

---------

Co-authored-by: Gustavo Antunes <17601467+gantunesr@users.noreply.github.com>
Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>
Co-authored-by: Plasma Corral <32695229+plasmacorral@users.noreply.github.com>
Co-authored-by: Nicolas MASSART <NicolasMassart@users.noreply.github.com>
Co-authored-by: Cal Leung <cleun007@gmail.com>
Co-authored-by: yande <110056475+Andepande@users.noreply.github.com>

gantunesr committed:
Merge PS

gantunesr committed:
Merge main

gantunesr committed:
Revert "Merge main"

This reverts commit 5e9cfcccccf7cbe04e1b0d88cb0951536d928fcf.

gantunesr committed:
Merge main

Cal-L committed:
Fix/4754 importing wallet via qr (#5634)

* Change ImportPrivateKey to TS file

* Add QR scanner screen to import stack

* Change ts to tsx

* Refactor import private key

* Don't show alert when importing account from account selector

* Fix deleting last imported account when in connected flow

* Fix dismissing keyboard when selecting top right nav button on webview

* Fix contrast for private key QR code

* Disable Permission System for WalletConnect and MMSDK

* Fix undefined origin issue

* fix browser title bar misalignment

* fix networkinfo avatar letter

* Resolve PR comments

---------

Co-authored-by: João Loureiro <joao.loureiro@consensys.net>

Cal-L committed:
Merge with main

jpcloureiro committed:
Add analytics to permission system (#5500)

* place analytics events

* rename permission connect request events

add source property

* update analytics

* fix import

* safeguarding when there are no active tabs

* rename event properties

* update source field

* update request completed properties

* fix networklist test

Cal-L committed:
Fix lint issues

cortisiko committed:
temporary bandaid for elements that do not appear.

gantunesr committed:
Merge main

gantunesr committed:
Merge main

gantunesr committed:
Merge branch 'feat/ledger-with-perm-system' of https://github.com/MetaMask/metamask-mobile into feat/ledger-integration

gantunesr committed:
FIX - Duplicated txs bugs

montelaidev committed:
fix: show imported label for accounts that are imported in the account overview

montelaidev committed:
fix: incorrect fromAddress param in MessageSign

montelaidev committed:
fix: change fromAddress to from

montelaidev committed:
bump keyring-controller

gantunesr committed:
Fix dependency bug & rebase

chrisd-ledger committed:
Feat/ledger integration (#5864)

* Rebuild Ledger account on PW reset

* Fix bluetooth pairing for android 12 and onward

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
fix merge issue

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
fix provider config in LedgerAccountInfo

montelaidev committed:
bump dependency

montelaidev committed:
fix #693

montelaidev committed:
fix #692

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
return null instead of empty string for accountTypeLabel

montelaidev committed:
fix personal sign to show ledger modal

montelaidev committed:
fix: show ledger modal for speedup and cancel tx

montelaidev committed:
show locked device error case

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
update yarn.lock

montelaidev committed:
fix lint

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
fix qr issue when scanning unknown account

montelaidev committed:
change info plist

montelaidev committed:
fix getLabelTextByAddress

montelaidev committed:
fix: cancel transaction if reject in mmm ui

montelaidev committed:
fix: #694 nonce too low

montelaidev committed:
Merge remote-tracking branch 'origin/main' into feat/ledger-integration

montelaidev committed:
refactor error and fix merge issue

montelaidev committed:
fix version requirement

gantunesr committed:
Update mnemonic util methods

gantunesr committed:
Update logic to retrieve mnemonic from KeyringController

gantunesr committed:
Add resolution for @ethereumjs/util

gantunesr committed:
Add @metamask/scure-bip39 module

gantunesr committed:
Bump KeyringController to v5.0.0

gantunesr committed:
Sort package.json

gantunesr committed:
Update yarn.lock

gantunesr committed:
Update import in ManualBackupStep2

gantunesr committed:
Update KeyringController patch

gantunesr committed:
Merge branch 'main' of https://github.com/MetaMask/metamask-mobile into bump-keyring-controller-v6

gantunesr committed:
Delete patch

gantunesr committed:
Bump keyring-controller to v6.0.0 and controller-utils to v3.3.0

gantunesr committed:
Update keyring-controller constructor

gantunesr committed:
Fix BaseControllerV2 change

gantunesr committed:
Add @metamask/eth-keyring-controller

gantunesr committed:
Add @metamask/eth-keyring-controller to declarations

gantunesr committed:
Add QRHardwareKeyring builder

gantunesr committed:
Subscribe to lock and unlock

gantunesr committed:
Fix exportSeedPhrase breaking change

gantunesr committed:
Update event subscription

gantunesr committed:
Merge main

gantunesr committed:
Merge main

gantunesr committed:
Update yarn.lock

gantunesr committed:
Add temporary solution to BackgroundBridge

gantunesr committed:
Solve merge conflicts

gantunesr committed:
Update hardware wallet connect route

gantunesr committed:
Merge branch 'main' into bump-keyring-controller-v6

gantunesr committed:
Merge main

gantunesr committed:
Bump KeyringController to v6.0.0

gantunesr committed:
Fix lint error

gantunesr committed:
Update subscription removal

gantunesr committed:
Update dependencies

gantunesr committed:
Add patch @metamask/eth-keyring-controller

stanleyyconsensys committed:
Feat/ledger integration (#7004)

* fix: change to display same error message with keystone

- use isHardwareAccount instead isQRHardwareAccount

* fix(#33): fix Unhandled exception on null object error in transaction

- remove duplicate function renderLedgerSignButton
- update latest renderLedgerSignButton function param for createStyles

* fix(#34): incorrect label when using browser

- fix action view default label , defaultProps is not able to translate

* fix(#24): fix uncatch error on unsigned transaction in ledger

- signLedgerTransaction is not being pass when using TransactionElement

* fix: remove label on transaction screen for MM accounts

* fix(#44): fix legacy code

- replace AppState.removeEventListener to remove
- replace TransactionController.cancelTransaction to ApprovalController.reject

* fix(#44): replace legacy approval transaction

- fix duplicate language block in en.json
- replace legacy approval transaction

* fix: ledger label on wallet screen

* fix(#44): remove legacy conflict code in transaction sending flow

---------

Co-authored-by: mlallai <matthieu.lallai@gmail.com>
Co-authored-by: stanleyyconsensys <stanley.yuen@consensys.net>

stanleyyconsensys committed:
fix: approval model missing navigate props (#7007)

stanleyyconsensys committed:
Merge branch 'feat/ledger-integration' into rebase-ledger

stanleyyconsensys committed:
chore: resolve conflict

stanleyyconsensys committed:
chore: update merge conflict

stanleyyconsensys committed:
chore: add back missing package

stanleyyconsensys committed:
chore: fix yarn.lock

stanleyyconsensys committed:
chore: fix yarn conflict and missing lib

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
fix: missing fontstyles

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

dawnseeker8 committed:
feat: Fix all lint error reported by ESlint (#7133)

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
fix: ledger label display incorrect style (#7146)

stanleyyconsensys committed:
fix: incorrect label type warning on account type tag (#7162)

- fix warning issue cause by incorrect prop type
- simplify account type checking on UI

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
fix: move ledger label logic to UI level (#7174)

ledger label determination should not place in component library level, as it relied on state, which component library should be stateless

stanleyyconsensys committed:
fix: handle exception when speedup or cancel transaction (#7176)

dawnseeker8 committed:
Feat: Fix the Message Sign issue for ledger devices (#7172)

* feat: add ledger sign message flow

- update message sign, personal sign and typed sign method
- implement ledger sign flow on above case

* fix: change isExternalHardwareAccount to use proper value

* fix: use org signature flow

- dismiss org sign model for ledger
- rollback deferSetAsSigned
- use org keyring sign flow

* fix: lint fix

* feat: Rebase ledger-intergation-main

add modified `yarn.lock` and `PersonalSign.tsx` files

* feat: backup a new version of personal sign

this version still have bug, need to fix later,

* feat: backup a new version of personal sign

this version still have bug, need to fix later,

* feat: refacotry the code and added missing patch

following has been done:
1. add two missing patch files from main.
2. refactory the RPCMethodMiddleware code to use common handle code.
3. fix the code errors in MessageSIgn.tsx and PersonalSign.tsx fils.

* feat: Refactory the code and refine the jsDoc

* feat: Rewrite the jsDoc

* feat: backup the working code before refactory

1. fix tsc error in reducer
2. fix the eth_sign missing logic due to mistake
3. based on stanely feedback, change some code based on his comments

* feat: refactory the RPCMethodMiddleware

1. Reimplement the generic code to middle method
2. dipatch actions in middleware method to reduce impact to actual RPC
3. create a whiteListedRPCs which can trigger actions.

this commit didnt include the reducer refactory
to make event action more generic.

* feat: give a default value to bypass lint error

* feat: Refactory signMesage reducer to rpcEvents

1. move whitelist code to reducer to centrally controlled.
2. rename `signMessage` actions and reducer to `rpcEvents`
3. change LedgerMessageSignModal ui to subscribe to different state obj

* feat: remove the redundant signMessage action

* feat: lint:fix and remove type OriginalRequest

---------

Co-authored-by: stanleyyuen <stanley.yuen@consensys.net>
Co-authored-by: Sébastien Van Eyck <sebastien.vaneyck@consensys.net>

stanleyyconsensys committed:
fix: Transaction Error when approve a token on ledger account (#7164)

dawnseeker8 committed:
Feat/fix lint tsc ledger (#7198)

* feat: Fix partially lint:tsc error in ledger code.

* feat: Backup of working version

Still have some bugs to fix.

* feat: Change to use `KeyringControllerMemState`

* feat: Remove redundant  unused import

* feat:Fix the keyringController param type mismatch

* feat: Fix the depcheck pipeline error

* feat: fix the missing property declared

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
fix: approve component migrate to use main branch logic (#7258)

stanleyyconsensys committed:
fix: incorrect ga tag (#7257)

stanleyyconsensys committed:
fix: uncatched ledger error when manual reject tx in dapp (#7274)

* fix: uncatched ledger error when manual reject tx in dapp

* fix: correct sendflow ledger transaction

dawnseeker8 committed:
Feat/fix unit test ledger main (#7222)

**Description**

Fix all unit tests failures from `ledger-integration-main` branch.
Create a draft PR to see whether CI can pass successfully.

stanleyyconsensys committed:
fix: update reject message transaction (#7289)

## **Description**
Update rejected message for ledger transaction in test dapp screen

## **Manual testing steps**

1. Pair Ledger
2. visit test dapp
3. create a token
4. after token deployed, click approve tokens
5. click Next when approval dialog show up
6. click Approve after next button click
7. when mobile app showing "confirm transaction by ...." 
8. click "Reject" on mobile app

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

igorms-cons committed:
Feat: add ledger error nonce too low (#7279)

## **Description**

The error occurred during a transaction from a Ledger account to a
normal account, after we manually decreased the nonce value on the
confirmation page.


1. pair ledger with imported SRP (secret recovery phrase) on MMM
2. turn on custom nonce in MMM settings>advanced
3. tap send from wallet view
4. transfer within your accounts
5. on confirmation, reduce nonce value by 2
6. confirm- approve on Ledger

Expected:- Error displays 'Nonce value too low' 


## **FIx**

We need to: 

**`app/components/hooks/useLedgerBluetooth.tsx`**
- Add a new value to LedgerCommunicationErrors.
- Inside the catch block of the processLedgerWorkflow function, add a
condition to set the error to the newly created one.

**`app/components/UI/LedgerModals/LedgerConfirmationModal.tsx`**
- Set error details with a dedicated title and subtitle, previously
added in local languages.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

stanleyyconsensys committed:
Merge branch 'main' into feat/ledger-intergtation-main

dawnseeker8 committed:
feat: fix all lint and unit tests after rebase (#7348)

## **Description**
Fix all lint and unit tests issue after rebase, regenerate all conflict
snapshop files.

stanleyyconsensys committed:
feat: add ledger keyring module in core

stanleyyconsensys committed:
Merge branch 'main' into feat/implement-ledger-keyring-into-mobile

stanleyyconsensys committed:
fix: conflict for v6 keyring controller

stanleyyconsensys committed:
fix: sync with main

stanleyyconsensys committed:
fix: sync with main

stanleyyconsensys committed:
fix: remove unuse package to sync with main

stanleyyconsensys committed:
fix: add patch for keyring controller to support getEthKeyringController

stanleyyconsensys committed:
chore: change ledger method to ledger library

stanleyyconsensys committed:
chore: update path

stanleyyconsensys committed:
fix: remove ledgerhq/metamask-keyring

stanleyyconsensys committed:
fix: ledger full update sync

dawnseeker8 committed:
feat: Commit the missing metamask-keyring library

dawnseeker8 committed:
feat: missing lock file in last commit

stanleyyconsensys committed:
fix: incorrect ledger type

dawnseeker8 committed:
feat: Fix some ledger keyring relative issue

stanleyyconsensys committed:
Revert "feat: Fix some ledger keyring relative issue"

This reverts commit a3533c474cf755e783244ff214afcd1b2a93fdb0.

stanleyyconsensys committed:
fix: update to use hardwaredevicename instead of keyring type for ledger

stanleyyconsensys committed:
fix: not to use keyring sign type data for ledger

stanleyyconsensys committed:
fix: to use self keyring package

igorms-cons committed:
update ledger_device const to  Ledger Hardware str

igorms-cons committed:
update with @igor-ms/ledgerhq-metamask-keyring

dawnseeker8 committed:
feat: Fix all lints errors and unit tests error (#7501)

## **Description**
Fix all lint error, tsc error and unit tests from
`implement-ledger-keyring-into-mobile`

Following have been done:
* Fix all lint errors
* Fix all tsc errors, temporately creating `SerializationOptions` and
`AccountDetails` type in Ledger.ts to bypass tsc type error, this will
be changed after new `metamask-keyring` library has been release.
* Fix all invalid account error due to incorrect account address such as
`0x0`, `0x1` etc in unit tests.
* Make all unit tests pass.

dawnseeker8 committed:
Merge branch 'main' into feat/implement-ledger-keyring-into-mobile

igorms-cons committed:
up for vivex test

igorms-cons committed:
revert last push

dawnseeker8 committed:
feat: fix the looping problem of permission (#7517)

## **Description**
Fix the looping error in
[`LedgerConfirmationModal.ts`](https://app.zenhub.com/workspaces/cet-metamask-mobile-ledger-integration-649aeabc8cdfe3878d13629f/issues/zh/87)

Following has been implemented:
* add a flag in `LedgerConfirmationModal.ts` to control whether to
rerender the screen.
* Modify the logic inside `onRetry()` to set the errorDetail to
undefined so that ledger screen can show up.
* Fix a edge case in `useBluetoothPermissions.ts` file when screen move
away from metamask to android setting, `bluetoothPermissionError`
haven't been set to undefined.

dawnseeker8 committed:
feat: Fix the screen looping in scan ledger device (#7537)

## **Description**
Fix the similar loop issue for scan devices screen when user disable the
`location` permisison in android.

dawnseeker8 committed:
Merge branch 'main' into feat/implement-ledger-keyring-into-mobile

gantunesr commented on app/util/address/index.js:
```diff
@@ -167,6 +211,43 @@ export function isQRHardwareAccount(address) {
+ * @param {String} address - String corresponding to an address
+ * @returns {String} - Returns address's i18n label text
+ */
+export function getLabelTextByAddress(address) {
+  if (!address) return null;
+  if (isHardwareAccount(address, [HardwareDeviceNames.ledger]))
+    return 'accounts.ledger';
+  if (isHardwareAccount(address, [KeyringTypes.qr]))
+    return 'transaction.hardware';
+  if (isImportedAccount(address)) return 'accounts.imported';
+
+  return null;
+}
```
There's some inconsistencies regarding Ledger and QR wallets. Would be good to double check that.

stanleyyconsensys replied:
as prev comment
Ledger keyring logic should not implement via keyring controller
hence, we havent touch the keyring controller and propose to add it via mobile directly
however, qr code still maintain in keyring controller for some reason
hence we adding a new type to map to the ledger keyring

gantunesr commented on app/util/hardwareWallet/signatureUtils.ts:
```diff
@@ -0,0 +1,42 @@
+
+  const navPromise = navMethodFactory.get(keyring.type as HardwareDeviceNames);
+
+  if (navPromise === undefined) {
+    //TODO: add error handling
+    throw new Error(`No nav type for address type ${keyring.type}`);
+  }
```
Improve error message

gantunesr commented on app/util/test/initial-background-state.json:
```diff
@@ -183,4 +183,4 @@
     "unapprovedTypedMessagesCount": 0
   }
-}
+}
```
Revert this change

gantunesr commented on ios/MetaMask/Info.plist:
```diff
@@ -43,6 +43,10 @@
 	<string>$(CURRENT_PROJECT_VERSION)</string>
 	<key>ITSAppUsesNonExemptEncryption</key>
 	<false/>
+	<key>LSApplicationQueriesSchemes</key>
+	<array>
+		<string>twitter</string>
+	</array>
```
Check why this was added

stanleyyconsensys replied:
i think this is duplicated, and should be remove

gantunesr commented on package.json:
```diff
@@ -558,4 +563,4 @@
       "ts-node>@swc/core": false
     }
   }
-}
+}
```
Revert this change

gantunesr commented on app/components/Views/LedgerConnect/hooks/useBluetoothPermissions.ts:
```diff
@@ -0,0 +1,105 @@
+
+      if (bluetoothAllowed) {
+        setHasBluetoothPermissions(true);
+        //Edge case, when `View setting` click, checkPermissions will trigger, and setBluetoothPermissionError(undefined), but when screen move away to android setting page,
```
nit
```suggestion
        // Edge case, when `View setting` click, checkPermissions will trigger, and setBluetoothPermissionError(undefined), but when screen move away to android setting page,
```

stanleyyconsensys replied:
resolved

gantunesr commented on app/constants/navigation/Routes.ts:
```diff
@@ -18,6 +18,19 @@ const Routes = {
+    CONNECT: 'ConnectHardwareWalletFlow',
+    SELECT_DEVICE: 'SelectHardwareWallet',
+    CONNECT_QR_DEVICE: 'ConnectQRHardwareFlow',
+    CONNECT_LEDGER: 'LedgerConnectFlow',
+    LEDGER_ACCOUNT: 'LedgerAccountInfo',
+  },
+  LEDGER_CONNECT_FLOW: {
+    ID: 'LedgerConnectFlow',
```
Duplicated values

stanleyyconsensys replied:
resolved by removing LEDGER_CONNECT_FLOW, instead merging into HW

gantunesr commented on app/components/UI/SignatureRequest/Root/Root.tsx:
```diff
@@ -32,13 +33,16 @@ const Root = ({
   const navigation = useNavigation();
   const { colors } = useTheme();
   const [showExpandedMessage, setShowExpandedMessage] = useState(false);
+  const visability = useSelector(
```
```suggestion
  const visibility = useSelector(
```

gantunesr replied:
Consider changing the variable name

stanleyyconsensys replied:
resolved

gantunesr commented on app/components/UI/SignatureRequest/Root/Root.tsx:
```diff
@@ -32,13 +33,16 @@ const Root = ({
   const currentPageMeta = messageParams?.meta;
 
-  if (!messageParams || !currentPageMeta || !approvalType) {
+  if (!messageParams || !currentPageMeta || !approvalType || !visability) {
```
```suggestion
  if (!messageParams || !currentPageMeta || !approvalType || !visibility) {
```

stanleyyconsensys replied:
resolved

gantunesr commented on android/.project:
```diff
@@ -1,17 +1,17 @@
-		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
-	</natures>
+        <name>MetaMask</name>
+        <comment>Project MetaMask created by Buildship.</comment>
```
Revert this change

stanleyyconsensys replied:
resolved

dawnseeker8 committed:
feat: fix the dedupe error in pipeline

dawnseeker8 committed:
Feat/fix forget devices issue (#7563)

## **Description**
Fix the `forget devices` in ledger didnt remove the ledger account in
account lists.

Following have been done:
* Implement new actions and reducer for accounts relative actions.
* modify `LedgerAccountInfo/index.tsx` `onForgetDevice()` event to
dispatch the account reload event.
* modify `accountSelector.tsx` to subscribe this actions and pass to
useAccounts() hook so that, hook can reload the account cache when
required.
* Fix some lint and implement some unit tests for relative codes.

dawnseeker8 committed:
Feat/display qr hardware in mobile (#7566)

## **Description**
Display the `QR hardware` for Keystone devices.

dawnseeker8 committed:
Merge branch 'main' into feat/implement-ledger-keyring-into-mobile

dawnseeker8 committed:
Feat/fix ledger UI feedback (#7582)

## **Description**

Fix zenhub problem that ledger `g` has been cut off from screen.
also fix the unit tests broken due to styling change.

## **Screenshot**

![image](https://github.com/MetaMask/metamask-mobile/assets/7315988/30916dd1-027c-4459-a5a5-0b29e49b124f)

dawnseeker8 committed:
Feat/ledger text center in pill (#7583)

## **Description**

Fix the styling issue of `ledger` text is not center of pill.




## **Screenshots/Recordings**

![image](https://github.com/MetaMask/metamask-mobile/assets/7315988/4df79dd8-fd28-4f62-baee-4faaaf9ee631)


![image](https://github.com/MetaMask/metamask-mobile/assets/7315988/a6ec7976-9fe8-4cf3-862e-11790216b761)

dawnseeker8 committed:
feat: Fix the bug from showing qr hardware (#7585)

## **Description**
Fix a issue of imported account still show `QR` Hardware.

## **Screenshots/Recordings**

![image](https://github.com/MetaMask/metamask-mobile/assets/7315988/7e52bc89-b395-440e-a8ac-925841b4479f)

gantunesr commented on app/component-library/components/Avatars/Avatar/foundation/AvatarBase/AvatarBase.test.tsx:
```diff
@@ -26,4 +26,19 @@ describe('AvatarBase', () => {
     );
     expect(wrapper).toMatchSnapshot();
   });
+
+  it('should render correctly without size', () => {
+    const blockieStyles = {
+      flex: 1,
+    };
+    const wrapper = shallow(
+      <AvatarBase>
+        <Image
+          source={{ uri: toDataUrl(DUMMY_IMAGE_DATA) }}
+          style={blockieStyles}
+        />
+      </AvatarBase>,
+    );
+    expect(wrapper).toMatchSnapshot();
+  });
```
Looks like this change should not be part of this PR

stanleyyconsensys replied:
resolved

stanleyyconsensys committed:
Fix/keyring controller patch (#7612)

## **Description**

The current keystone hardware in ledger branch, when user click forget
devices, the screen will output Cannon read property ‘removedAccounts’
of undefined. error message.
This patch fix will fix the issue.

## **Related issues**


https://app.zenhub.com/workspaces/cet-metamask-mobile-ledger-integration-649aeabc8cdfe3878d13629f/issues/zh/98


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it
is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

stanleyyconsensys committed:
Fix/resolve ledger pr comment (#7613)

## **Description**

fix PR comment from this PR
https://github.com/MetaMask/metamask-mobile/pull/7385/

## **Related issues**

Fixes: #

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

stanleyyconsensys commented on app/core/BackgroundBridge/BackgroundBridge.js:
```diff
@@ -155,10 +159,14 @@ export class BackgroundBridge extends EventEmitter {
+      });
+    } catch (e) {
+      // Do nothing
+    }
```
resolved

stanleyyconsensys commented on app/components/Nav/Main/RootRPCMethodsUI.js:
```diff
@@ -203,7 +209,31 @@ const RootRPCMethodsUI = (props) => {
-        Engine.acceptPendingApproval(transactionMeta.id);
+
+        // eslint-disable-next-line no-console
+        console.log(transactionMeta);
```
resolved

stanleyyconsensys commented on app/components/Nav/App/index.js:
```diff
@@ -574,6 +579,29 @@ const App = ({ userLoggedIn }) => {
     </Stack.Navigator>
   );
 
+  // eslint-disable-next-line react/prop-types
```
not necessary, removed

stanleyyconsensys commented on app/component-library/components/Avatars/Avatar/foundation/AvatarBase/__snapshots__/AvatarBase.test.tsx.snap:
@dawnseeker8 can we remove this?

stanleyyconsensys commented on ios/MetaMask.xcodeproj/project.pbxproj:
why we need to update this file?

stanleyyconsensys commented on ios/Podfile.lock:
```diff
@@ -930,6 +952,6 @@ SPEC CHECKSUMS:
-PODFILE CHECKSUM: 8d3adf4eeb9c38c94e164bf39d459809e3147413
+PODFILE CHECKSUM: a75a7b4687a558d94615247d0e8a266e714632e5
 
-COCOAPODS: 1.12.1
```
why we need to change the version

stanleyyconsensys committed:
fix: resolve comment (#7627)

## **Description**

- remove blank line
- revert back escape line
- revert cocopod version
- remove duplicated info.plist
- remove EXCLUDED_ARCHS from project
- relocate the ledger bluetooth hooks


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it
is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

dawnseeker8 committed:
Merge branch 'main' into feat/implement-ledger-keyring-into-mobile

dawnseeker8 committed:
feat: Fix snapshot unit tests failure after rebase

stanleyyconsensys committed:
feat: add metric for swap (#7630)

## **Description**
 - add account type to swap metric 

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it
is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

igorms-cons committed:
Feat/ledger metrics (#7622)

## **Description**

Add Metametrics to Ledger use cases



Fixes: #
Update files


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Monte Lai <monte.lai@consensys.net>

dawnseeker8 committed:
Merge branch 'main' into feat/implement-ledger-keyring-into-mobile

dawnseeker8 committed:
Merge branch 'main' into feat/implement-ledger-keyring-into-mobile

gantunesr commented:
Closed in favor of https://github.com/MetaMask/metamask-mobile/pull/7757
